src/std/srfi/srfi-19.scm

```diff
@@ -253,7 +258,25 @@
 ;;; the time structure; creates the accessors, too.
 ;;; wf: changed to match srfi documentation. uses mzscheme structures & inspectors
 ;;; vyzo: change to use gerbil defstruct
-(defstruct time (type nanosecond second))
+;;; fare: added checking in make-time
+(defstruct time (type nanosecond second)
+  constructor: init!)
+
+(defmethod {init! time}
```

Comment:
Do we really want the type checking in the constructor? 
it is exericsed in blanket fashion, which will result in slowdown when constructing time objects (and we do that potentially a lot).

Reply:
Well, initially I wanted to rename the constructor `%make-time` and have the exported `make-time` function be a wrapper around that, whereas most internal functions would use `%make-time`. Unhappily, `defstruct` doesn't seem to allow that.

I wanted the exported function `make-time` to have checks, which would have caught my stupid error from earlier this week.

Reply:
Agreed on public api having chceks; just export rename it to make-time in 19.ss